examples/silentpayments.c

```diff
@@ -0,0 +1,619 @@
+/*************************************************************************
+ * To the extent possible under law, the author(s) have dedicated all    *
+ * copyright and related and neighboring rights to the software in this  *
+ * file to the public domain worldwide. This software is distributed     *
+ * without any warranty. For the CC0 Public Domain Dedication, see       *
+ * EXAMPLES_COPYING or https://creativecommons.org/publicdomain/zero/1.0 *
+ *************************************************************************/
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <secp256k1_extrakeys.h>
+#include <secp256k1_silentpayments.h>
+
+#include "examples_util.h"
+
+/* Use my_memcmp_var instead of memcmp.
+ *
+ * Normally, memcmp should be fine, but we use my_memcmp_var
+ * here to avoid a false positive from valgrind on macOS.
+ * TODO: remove this in the event the bug is fixed with valgrind in the future.
+ */
+static int my_memcmp_var(const void *s1, const void *s2, size_t n) {
+    const unsigned char *p1 = s1, *p2 = s2;
+    size_t i;
+
+    for (i = 0; i < n; i++) {
+        int diff = p1[i] - p2[i];
+        if (diff != 0) {
+            return diff;
+        }
+    }
+    return EXIT_SUCCESS;
```

Comment:
in 070ea002339bf10603da5dd772795c7ccae3af51, function `my_memcmp_var`
```suggestion
    return 0;
```
as this doesn't indicate a process execution status